lib/components/Pagination/index.tsx

```diff
@@ -30,8 +31,12 @@ export const Pagination = ({
     });
 
     useEffect(() => {
-        onClickPagination({ currentPage, totalPageCount });
-    }, [currentPage, onClickPagination, totalPageCount]);
+        if (initialPage) setCurrentPage(initialPage);
+    }, [initialPage]);
+
+    useEffect(() => {
+        onClickPagination?.({ currentPage, totalPageCount });
+    }, [currentPage, totalPageCount]);
```

Comment:
do we want to trigger onClickPagination with totalPageCount changes too?

Reply:
Good point, I don't think we should trigger. Will remove